crates/compiler/test_gen/src/gen_panic.rs

```diff
@@ -0,0 +1,69 @@
+use indoc::indoc;
+
+#[cfg(feature = "gen-llvm")]
+use crate::helpers::llvm::expect_runtime_error_panic;
+
+#[cfg(feature = "gen-wasm")]
+use crate::helpers::wasm::expect_runtime_error_panic;
+
+#[test]
+#[cfg(any(feature = "gen-llvm", feature = "gen-wasm"))]
+#[should_panic = r#"User crash with message: "hello crash""#]
+fn crash_literal() {
+    expect_runtime_error_panic!(indoc!(
+        r#"
+        app "test" provides [main] to "./platform"
+
+        main = if Bool.true then crash "hello crash" else 1u8
+        "#
+    ));
+}
+
+#[test]
+#[cfg(any(feature = "gen-llvm", feature = "gen-wasm"))]
+#[should_panic = r#"User crash with message: "hello crash""#]
+fn crash_variable() {
+    expect_runtime_error_panic!(indoc!(
+        r#"
+        app "test" provides [main] to "./platform"
+
+        main =
+            msg = "hello crash"
+            if Bool.true then crash msg else 1u8
+        "#
+    ));
+}
+
+#[test]
+#[cfg(any(feature = "gen-llvm", feature = "gen-wasm"))]
+#[should_panic = r#"User crash with message: "turns out this was fallible""#]
+fn crash_in_call() {
+    expect_runtime_error_panic!(indoc!(
+        r#"
+        app "test" provides [main] to "./platform"
+
+        getInfallible = \result -> when result is
+            Ok x -> x
+            _ -> crash "turns out this was fallible"
+
+        main =
+            x : [Ok U64, Err Str]
+            x = Err ""
+            getInfallible x
+        "#
+    ));
+}
+
+#[test]
+#[cfg(any(feature = "gen-llvm"))]
+// wasm: blocked on "compiling function underran the stack" in wasm3
```

Comment:
@brian-carroll could use some help here. it's not blocking but just fyi. the problem seems to be that wasm3 wants the stack to appear polymorphic after the crash instruction, but it already should be so because we emit an UNREACHABLE. not sure what's going on, could use a pointer

Reply:
> it already should be so because we emit an UNREACHABLE

I agree! That's what I would have thought too!
OK I will pull the branch and see if I can get anywhere.

Reply:
Could a good idea if you let me know what kind of things you already tried to debug it.
Then I can tell you if I did anything different.

Reply:
I'm using the debug HTML page crates/compiler/test_gen/src/helpers/debug-wasm-test.html
Chrome's error message is a bit more helpful:

> Uncaught (in promise) CompileError: WebAssembly.instantiate(): Compiling function #901 failed: not enough arguments on the stack for i32.store8 (need 1, got 0)

Reply:
Understood, thanks Brian. I remember debugging previously but forgot about this case. Too many things to keep in mind, I suppose. I think we should just remove this test macro and force always using `assert_evals_to`. Good to delete code anyway